compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -678,17 +691,16 @@ private void generateBinaryCompareIns(BIRNonTerminator.BinaryOp binaryIns, int o
         this.storeToVar(binaryIns.lhsOp.variableDcl);
     }
 
-    private String getDecimalCompareFuncName(int opcode) {
-
+    private String getCompareFuncName(int opcode) {
         switch (opcode) {
             case IFGT:
-                return "checkDecimalGreaterThan";
+                return "compareValueGreaterThan";
             case IFGE:
-                return "checkDecimalGreaterThanOrEqual";
+                return "compareValueGreaterThanOrEqual";
             case IFLT:
-                return "checkDecimalLessThan";
+                return "compareValueLessThan";
             case IFLE:
-                return "checkDecimalLessThanOrEqual";
+                return "compareValueLessThanOrEqual";
```

Comment:
We can remove `checkDecimalGreaterThan`, `checkDecimalGreaterThanOrEqual`, `checkDecimalLessThan` and `checkDecimalLessThanOrEqual` functions from the typechecker. right?